doc/ref/plot_options/data.ipynb

```diff
@@ -45,7 +61,224 @@
+    "## `group_label`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `group_label` option lets you set a custom name for the key dimension that distinguishes multiple series in an overlay plot. When your data contains multiple groups, hvPlot creates an overlay plot where each series is identified by a key dimension. By default, this key is labeled “Variable,” but you can override it with a more descriptive name using `group_label`. This is especially useful when the grouping variable has a clear meaning, such as geographical coordinates or other numeric identifiers."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import hvplot.pandas  # noqa\n",
+    "\n",
+    "df = pd.DataFrame([[1, 2], [3, 4], [5, 6]], columns=['x', 'y'])\n",
+    "\n",
+    "df.hvplot.line(group_label='Category')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example, setting `group_label='Category'` customizes the legend to display this label instead of the default, making the plot more informative."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## `kind`\n",
+    "The kind option determines the type of plot to generate from your data. By specifying a plot kind (such as ‘line’, ‘scatter’, or ‘bar’), you tell hvPlot which plot to create. The default is 'line', which generates a line plot. Changing the `kind` parameter lets you quickly experiment with different visual representations without altering your data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import hvplot.pandas  # noqa\n",
+    "\n",
+    "df = pd.DataFrame({\n",
+    "    'year': [2018, 2019, 2020, 2021],\n",
+    "    'sales': [150, 200, 250, 300]\n",
+    "})\n",
+    "\n",
+    "line_plot = df.hvplot(x='year', y='sales', width=600, title=\"Default line plot\")\n",
+    "bar_plot = df.hvplot(x='year', y='sales', width=600, kind='bar', title=\"Bar plot\")\n",
+    "\n",
+    "line_plot + bar_plot"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example, the first plot uses the default (`kind='line'`), while the second explicitly sets `kind='bar'` to create a bar chart. You can also specify it as an attribute of the `hvplot` class:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df.hvplot.bar(x='year', y='sales')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## `label`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The label option allows you to specify a custom name for your dataset that appears in the plot title or legend."
```

Comment:
There's the `title` plot option that has a somewhat similar effect. It'd be nice to explain when `label` should be used over `title`.

Reply:
I'm not sure yet what the answer to this is. Will look into it more...